ElementX/Sources/Screens/RoomScreen/View/RoomScreen.swift

```diff
@@ -75,10 +75,21 @@ struct RoomScreen: View {
         }
     }
     
+    @ToolbarContentBuilder
     var toolbar: some ToolbarContent {
-        ToolbarItem(placement: .navigationBarLeading) {
+        // .principal + .primaryAction works better than .navigation leading + trailing
+        // as the latter disables interaction in the action button for rooms with long names
+        ToolbarItem(placement: .principal) {
             RoomHeaderView(context: context)
         }
+        
+        ToolbarItem(placement: .primaryAction) {
+            Button {
+                context.send(viewAction: .displayRoomDetails)
+            } label: {
+                Image(systemName: "ellipsis.circle")
+            }
+        }
```

Comment:
As discussed could maybe do this to use the overflow menu as provided by the system.
```suggestion
        ToolbarItem(id: "RoomDetailsAction", placement: .secondaryAction, showsByDefault: false) {
            Button {
                context.send(viewAction: .displayRoomDetails)
            } label: {
                Label("Room Details", systemImage: "info.circle")
            }
        }
```